components/infobox/wikis/splatoon/infobox_league_custom.lua

```diff
@@ -0,0 +1,98 @@
+---
+-- @Liquipedia
+-- wiki=splatoon
+-- page=Module:Infobox/League/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+local Logic = require('Module:Logic')
+local String = require('Module:StringUtils')
+local Variables = require('Module:Variables')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local League = Lua.import('Module:Infobox/League', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Title = Widgets.Title
+
+local CustomLeague = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _game
+
+local _GAME = mw.loadData('Module:GameVersion')
+
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_args = league.args
+
+	league.addToLpdb = CustomLeague.addToLpdb
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+	league.liquipediaTierHighlighted = CustomLeague.liquipediaTierHighlighted
+
+	return league:createInfobox(frame)
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'sponsors' then
+		table.insert(widgets, Cell{name = 'Official Device', content = {_args.device}})
+	elseif id == 'gamesettings' then
+		return {
+			Cell{name = 'Game version', content = {
+					CustomLeague._getGameVersion()
+				}},
+			}
+	elseif id == 'customcontent' then
+		if _args.player_number then
+			table.insert(widgets, Title{name = 'Players'})
+			table.insert(widgets, Cell{name = 'Number of players', content = {_args.player_number}})
+		end
+
+		--teams section
+		if _args.team_number then
+			table.insert(widgets, Title{name = 'Teams'})
+			table.insert(widgets, Cell{name = 'Number of teams', content = {_args.team_number}})
+		end
+	end
+	return widgets
+end
+
+function CustomLeague:liquipediaTierHighlighted()
+	return Logic.readBool(_args['splatoonpremier'])
+end
```

Comment:
Is this used? Otherwise remove (and if so, from `addToLpdb` and `defineCustomPageVariables` as well.

@hjpalpha should we add a default that's something like `Logic.readBool(_args['premier'])`?

Reply:
They are currently for some of the world championship pages
https://liquipedia.net/splatoon/Splatoon_2_World_Championship/2019 

(If all three mentioned areas were moved, the yellow tier would not highlight at all) (both on infobox and portal)
![image](https://user-images.githubusercontent.com/88981446/195320507-1dd9c451-a7c0-4c44-89c8-b454e04ff3e4.png)

Reply:
Can the wiki code input be `publisherpremier` instead?

Reply:
> Is this used? Otherwise remove (and if so, from `addToLpdb` and `defineCustomPageVariables` as well.
> 
> @hjpalpha should we add a default that's something like `Logic.readBool(_args['premier'])`?

okay for me
(sc2 would just not set it as true)

Reply:
> Can the wiki code input be `publisherpremier` instead? @Hesketh2

yes I think that's doable. Since the highlighted tier dont used that often as well
just replaced **splatoonpremier** with **publisherpremier** ?